frontend/src/app/AppNotificationDrawer.tsx

```diff
@@ -77,7 +81,27 @@ const AppNotificationDrawer: React.FC<AppNotificationDrawerProps> = ({ onClose }
                 <NotificationDrawerListItemBody
                   timestamp={calculateRelativeTime(notification.timestamp, currentTime)}
                 >
-                  {notification.message}
+                  <Stack hasGutter>
+                    <StackItem>{notification.message}</StackItem>
+                    {notification.actions?.length && (
```

Comment:
Pretty sure you can't do this... empty arrays used like `array.length && renderCode` will render 0 in the DOM when they are empty. (this is from memory, not tested)

Reply:
I'm glad you don't miss anything :)